upload/src/addons/SV/WarningImprovements/Entity/WarningCategory.php

```diff
@@ -124,6 +124,16 @@ protected function _postDelete()
             }
         }
 
+        foreach ($this->WarningDefinitions as $warningDefinition)
+        {
+            $warningDefinition->delete();
```

Comment:
Should ensure the custom wasn't isn't deleted.

If this category is being deleted, then move the custom warning to the first category.

If this is the last category, reject the delete in `_preDelete()` with a relevant error message that you can't delete the last category. The WarningDefinition should also reject deletes for custom warning in the `_preDelete()` method too.

Reply:
`isDeletable()` looks incomplete, and the custom warning needs to be re-assigned to a different category.